src/xml_generator.jl

```diff
@@ -0,0 +1,263 @@
+          str_cond = pnet.transitions[i].condition
+          str = string(out_place_list[j][1].name, " = $(str_cond);\n") 
+          out_port_list = out_port_list * str
+        else
+          str = string(out_place_list[j][1].name, " = __output[$k][0];\n") 
+          out_port_list = out_port_list * str
+        end
+      end
+      code = new_child(mod, "code")
+      add_cdata(xpnet, code, string("std::vector<std::vector<std::string>> __output = zeda::execute(implementation_$i, {$(in_str)}, $(num_outs));\n", out_port_list))
+    end
+
+    if !isempty(pnet.transitions[i].condition)
+      cond = new_child(trans, "condition")
+      add_text(cond, pnet.transitions[i].condition)
+    end
+
+    for j in 1:length(pnet.arcs)
+      if pnet.arcs[j].transition == pnet.transitions[i]
+        if pnet.arcs[j].type == :out_many
+          connect_prt = new_child(trans, "connect-out-many")
+          set_attributes(connect_prt, Dict("port"=>pnet.arcs[j].place.name, "place"=>string(pnet.arcs[j].place.name)))
+        else
+          connect_prt = new_child(trans, string("connect-", pnet.arcs[j].type))
+          set_attributes(connect_prt, Dict("port"=>pnet.arcs[j].place.name, "place"=>string(pnet.arcs[j].place.name)))
+        end
+      end
+    end
+
+    impl_port_connect = new_child(trans, "connect-in")
+    set_attributes(impl_port_connect, Dict("port"=>"implementation_$i", "place"=>"implementation_$i"))
+  end
+  return xpnet
+end
+
+struct XML_workflow
+  name::String
+  path::String
+end
+
+
+"""
+    workflow_generator(pnet::PetriNet)
+    workflow_generator(pnet::PetriNet, path::String)
+Given a Petri net description, creates an XML workflow and writes it to a file in the path.
+
+# Examples
+```julia-repl
+julia> pn = PetriNet("hello_julia")
+A Petri net with name "hello_julia", having 0 ports, 0 places, and 0 transitions.
+
+
+julia> p1 = place("input1", :string)
+Place input1 with control token created.
+
+
+julia> p2 = place("input2",:string)
+Place input2 with control token created.
+
+
+julia> p3 = place("output",:string)
+Place output with control token created.
+
+
+julia> t = transition("trans")
+Transition trans created.
+
+
+julia> connect(pn, p1,t, :in)
+A Petri net with name "hello_julia", having 0 ports, 1 places, and 1 transitions.
+
+
+julia> connect(pn, p2,t, :read)
+A Petri net with name "hello_julia", having 0 ports, 2 places, and 1 transitions.
+
+
+julia> connect(pn, p3,t, :out_many)
+A Petri net with name "hello_julia", having 0 ports, 3 places, and 1 transitions.
+
+
+julia> connect(pn, p1, :in)
+A Petri net with name "hello_julia", having 1 ports, 3 places, and 1 transitions.
+
+
+julia> connect(pn, :in, p2)
+A Petri net with name "hello_julia", having 2 ports, 3 places, and 1 transitions.
+
+
+julia> connect(pn, :out, p3)
+A Petri net with name "hello_julia", having 3 ports, 3 places, and 1 transitions.
+
+
+julia> workflow_generator(pn)
+"An XML workflow "hello_julia" has been written to the location: /root/tmp."
```

Comment:
Maybe replace `/root/` with `/home/user/`

Reply:
the message string is auto generated by Julia. But I can change the location to store the generated image. That way the message also changes.